compiler/crates/relay-typegen/src/lib.rs

```diff
@@ -142,7 +149,8 @@ impl<'schema, 'config> TypeGenerator<'schema, 'config> {
         let input_variables_type = self.generate_input_variables_type(typegen_operation);
 
         let selections = self.visit_selections(&typegen_operation.selections);
-        let mut response_type = self.selections_to_babel(selections, false, None);
+        let mut response_type =
+            self.selections_to_babel(Some(typegen_operation.type_), selections, false, None);
```

Comment:
Haha, just noticed we still have the `babel` name in here from my original 1:1 translation from the JS code. This is some of the most confusing part of the compiler to me, so it was just a direct line-by-line translation where possible…

Reply:
Yeah, same. As mentioned, I made these same changes earlier in Flow (and TypeScript before that), but it's not a function that's easy to read. In any case, I renamed it to `selections_to_ast` to get rid of the Babel name.